File: Frontend/parth-traders/src/app/user/core/models/OrderDetail.ts
import { Product } from "./Product";

export class OrderDetail {
    productSku: string;
    price: number;
    quantity: number;
    discount: number;
    totalPrice: number;

    constructor(product: Product, quantity: number, totalPrice: number) {
        this.productSku = product.productSku;
        this.price = product.unitPrice;
        this.discount = product.discount;
        this.quantity = quantity;
        this.totalPrice = totalPrice;
    }

    incrementProductCount() {
        this.quantity += 1;
        this.totalPrice += this.price;
    }

    decrementProductCountFrom() {
        this.quantity -= 1;
        this.totalPrice -= this.price;
    }

    getProductQuantity() {
        return this.quantity;
    }

    getTotalPrice() {
        return this.totalPrice;
    }
}

File: Backend/Parth Traders.Domain/Entity/User/OrderDetail.cs
using Parth_Traders.Domain.Entity.Admin;

namespace Parth_Traders.Domain.Entity.User
{
    public class OrderDetail
    {
        public long OrderDetailId { get; set; }

        public virtual Order Order { get; set; }

        public virtual Product Product { get; set; }

        public long Price { get; set; }

        public long Quantity { get; set; }

        public long Discount { get; set; }

        public long TotalPrice { get; set; }
    }
}
